projects/plugins/jetpack/extensions/blocks/top-posts/top-posts.php

```diff
@@ -46,19 +51,15 @@ function load_assets( $attributes ) {
 	/*
 	 * We cannot rely on obtaining posts from the block because
 	 * top posts might have changed since then. As such, we must
-	 * make another request to check for updated stats.
+	 * check for updated stats.
 	 */
-	$request_url = sprintf(
-		'/wp-json/wpcom/v2/top-posts?period=%1$s&number=%2$s&types=%3$s',
-		$attributes['period'],
-		$attributes['postsToShow'],
-		implode( ',', array_keys( array_filter( $attributes['postTypes'] ) ) )
-	);
+	$period = $attributes['period'];
+	$number = $attributes['postsToShow'];
+	$types  = implode( ',', array_keys( array_filter( $attributes['postTypes'] ) ) );
 
-	$request = wp_remote_get( home_url( $request_url ) );
-	$data    = json_decode( wp_remote_retrieve_body( $request ), true );
+	$data = Jetpack_Top_Posts_Helper::get_top_posts( $period, $number, $types );
```

Comment:
The `context` of Each item from `$data` is type of `WP_Term` now because it doesn't go thru the serialization and serialization processes, so L104 is failing. I guess we could either force covert `context` to array or access it as an object. The related error is like the following,

```
[06-Dec-2023 21:19:01 UTC] PHP Fatal error:  Uncaught Error: Cannot use object of type WP_Term as array in /usr/local/src/jetpack-monorepo/projects/plugins/jetpack/extensions/blocks/top-posts/top-posts.php:104
Stack trace:
#0 /var/www/html/wp-includes/class-wp-block.php(258): Automattic\Jetpack\Extensions\Top_Posts\load_assets()
#1 /var/www/html/wp-includes/blocks.php(1484): WP_Block->render()
#2 /var/www/html/wp-includes/blocks.php(1522): render_block()
#3 /var/www/html/wp-includes/class-wp-hook.php(324): do_blocks()
#4 /var/www/html/wp-includes/plugin.php(205): WP_Hook->apply_filters()
#5 /var/www/html/wp-includes/post-template.php(256): apply_filters()
#6 /var/www/html/wp-content/themes/twentytwentyone/template-parts/content/content-single.php(23): the_content()
#7 /var/www/html/wp-includes/template.php(792): require('...')
#8 /var/www/html/wp-includes/template.php(725): load_template()
#9 /var/www/html/wp-includes/general-template.php(206): locate_template()
#10 /var/www/html/wp-content/themes/twentytwentyone/single.php(18): get_template_part()
#11 /var/www/html/wp-includes/template-loader.php(106): include('...')
#12 /var/www/html/wp-blog-header.php(19): require_once('...')
#13 /var/www/html/index.php(17): require('...')
#14 {main}
  thrown in /usr/local/src/jetpack-monorepo/projects/plugins/jetpack/extensions/blocks/top-posts/top-posts.php on line 104
```